src/common/schema-validation.ts

```diff
@@ -12,22 +13,39 @@ export type JSONSchemaTypeWithId<T> = JSONSchemaType<T> & { $id: string };
  *
  * @see https://ajv.js.org/guide/managing-schemas.html#pre-adding-all-schemas-vs-adding-on-demand
  */
-export const getAjvSchema = <T>(
+const getAjvSchema = <T>(
 	schema: JSONSchemaTypeWithId<T>,
 ): ValidateFunction<T> => {
-	return ajv.getSchema(schema.$id) ?? ajv.compile(schema);
+	return (
+		schemaValidation.getSchema(schema.$id) ?? schemaValidation.compile(schema)
+	);
 };
 
-export function assertSchema<T>(
+export type JSONSchemaTypeWithId<T> = JSONSchemaType<T> & { $id: string };
+
+export function validateSchema<T>(
```

Comment:
**note:** Adds `validateSchema` to enable schema validation error mapping (as it is required with `assertSchema`).